docs/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,98 @@
+
+This is accomplished by assigning a Build ID of your choice to the set of code that constitutes a Worker, and specifying whether these versions are compatible or not with one another via changes to the aforementioned graph.
+
+### When and what should you use Worker Versioning for?
+
+The primary use case that this feature makes much simpler is the deployment of incompatible changes to short-lived workflows. On Task Queues making use of this feature, the workflow starter does not need to know that there are new versions being introduced. New [Workflow Executions](/concepts/what-is-a-workflow-execution) will run using the new code in the newly deployed workers, and old (but still open) workflows will only be processed by workers with an appropriate version. Old workers may be decommissioned once there are no longer any open workflows using their version.
+
+For example if you have a workflow which never takes longer than a day to execute, a great strategy is to simply always assign a new Build ID to every new build of your worker and add it to the graph as the new default version. Because you know your workflow never takes more than a day to complete, you know that you'll never have to leave older workers running for more than a day once you've deployed the new version (assuming availability).
+
+It's perfectly fine to do this with longer-lived workflows as well. You'll just need to be aware that you may need to keep multiple different versions of workers running while open workflows complete.
+
+You can also use the feature to prevent a buggy codepath from being taken on currently open workflows. You can do this by adding a new
+version to the graph, and specifying it as _compatible_ with an existing version you would like to ensure no future Workflow Tasks execute on. Because the new version will process existing histories, it is subject to the normal [determinism constraints](/concepts/what-is-a-workflow-definition#non-deterministic-change), and you might need to use one of the [versioning apis](/concepts/what-is-a-workflow-definition#workflow-versioning).
+
+### How to use Worker Versioning
+
+To make use of this feature, you'll need to:
+
+1. Define a Worker Build ID version graph for the Task Queue you are concerned with:
+   You can use either `tctl` or your SDK of choice. TODO: Links
+2. Enable the feature on your worker by specifying a Build ID:
+   TODO: Links for each SDK's worker options
+
+#### Defining the version graph
+
+Whether you use `tctl` or an SDK, updating the version graph feels the same.
+
+You'll specify the task queue you're targeting, the Build ID you're adding (or moving), whether or not it is to become the new default version, and some existing version it should be considered compatible with, if any.
```

Comment:
They're effectively synonyms for the user. I think there is some value in using both terms here, distinguished like you said, since it helps contextualize when we're talking about something inside vs outside of the graph... but I'm open to suggestion.

Maybe I can just add a little bit about the two terms near the top?